tests/operator_tests.py

```diff
@@ -476,13 +467,79 @@ def test_pda_power(self):
         ex = np.power(n, [2, 3, 4])
         self.assertListEqual(p.to_list(), ex.tolist())
 
+        # Test a singleton with and without a Boolean argument
+        n = np.array([7])
+        a = ak.array(n)
+
+        p_True = ak.power(a, 3, True)
+        p_Imply = ak.power(a, 3)
+        p_False = ak.power(a, 3, False)
+        self.assertListEqual(p_True.to_list(), p_Imply.to_list())
+        self.assertListEqual(p_False.to_list(), a.to_list())
+
+        # Test an with and without a Boolean argument, all the same
+        n = np.array([0, 0.0, 1, 7.0, 10])
+        a = ak.array(n)
+
+        truth_True = ak.array([True, True, True, True, True])
+        truth_False = ak.array([False, False, False, False, False])
+        p_True = ak.power(a, 3, truth_True)
+        p_Imply = ak.power(a, 3)
+        p_False = ak.power(a, 3, truth_False)
+        self.assertListEqual(p_True.to_list(), p_Imply.to_list())
+        self.assertListEqual(p_False.to_list(), a.to_list())
+
+        # Test a singleton with and without a Boolean argument, mixed
+        n = np.arange(10)
+        a = ak.array(n)
+        n_Calculated = np.array([0, 1, 4, 3, 16, 5, 36, 7, 64, 9])
+        a_Calculated = ak.array(n_Calculated)
+
+        truth = ak.array([True, False, True, False, True, False, True, False, True, False])
+        p = ak.power(a, 2, truth)
+        self.assertListEqual(p.to_list(), a_Calculated.to_list())
+
+        # Test invalid input, negative
         n = np.array([-1.0, -3.0])
         a = ak.array(n)
 
         p = ak.power(a, 0.5)
         ex = np.power(n, 0.5)
         self.assertTrue(np.allclose(p.to_ndarray(), ex, equal_nan=True))
 
+        # Test invalid input, INF
+        I = np.Infinity
+        n = np.array([I, -I])
+        a = ak.array(n)
+        n_Calculated = np.array([I, I])
+        a_Calculated = ak.array(n_Calculated)
+
+        p = ak.power(a, 2)
+        self.assertListEqual(p.to_list(), a_Calculated.to_list())
```

Comment:
we should compare directly against the numpy array here instead of converting to a pdarray first to verify we're handling `inf` the same way they are

```python
n = np.array([np.inf, -np.inf])
a = ak.array([np.inf, -np.inf])
self.assertListEqual(np.power(n, 2).tolist(), ak.power(a, 2).to_list())
```